src/frontend/components/app/modal-info.tsx

```diff
@@ -0,0 +1,43 @@
+import { Modal } from '@adminjs/design-system'
+import React from 'react'
+import { useSelector } from 'react-redux'
+import { ModalButton } from '../../interfaces'
+import { ReduxState } from '../../store'
+import { useModal, useTranslation } from '../../hooks'
+
+export const ModalInfo = () => {
+  const rawModal = useSelector((state: ReduxState) => state.modal)
+  const { closeModal } = useModal()
+  const { translateButton, translateLabel, translateMessage } = useTranslation()
+  const modal = { ...rawModal }
+  const close = () => {
+    closeModal()
+  }
+  const confirm = () => {
+    closeModal()
+    modal.confirmAction()
+  }
+  const buttons: Array<ModalButton> = []
+  if (modal.type === 'confirm' && typeof modal.confirmAction === 'function') {
+    buttons.push({ label: translateButton('Cancel', modal.resourceId), variant: 'default', onClick: close })
+    buttons.push({ label: translateButton('Confirm', modal.resourceId), variant: 'primary', onClick: confirm })
```

Comment:
I wonder if it's possible to add custom buttons using `openModal`? People can have different use cases for modals, not just confirming an action

Reply:
This solution was aimed to replace alert and confirm action in very specific conditions, it can be used in other places of course but some elements are fixed. If someone need more custom solution he/she could use generic Modal component from design-system

Reply:
Then I think it should be named `ConfirmModal` and `useConfirmModal` hook because `Modal` and `useModal` sound like generic elements

Reply:
changed

Reply:
The generic modal will be continued in AJS-415